Add scheduler name support - [Update #990]

bbsbb opened the pull request:
Hey there,

This is an updated, rebased continuation of Micah Coletti's PR allowing specification of an alternative scheduler.

If there is additional work that needs to be done on this to meet a standard deserving of merging, I'm up for it.

Would really appreciate involvement and direct answer if this is something that could become part of the postgres operator as it is a blocking issue for us.

Thank you for the great work on the project!

Jan-M commented:
:+1:

FxKu commented on pkg/apis/acid.zalan.do/v1/crds.go:
```diff
@@ -569,6 +569,9 @@ var PostgresCRDResourceValidation = apiextv1.CustomResourceValidation{
 							},
 						},
 					},
+					"schedulerName": {
+						Type: "string",
+					},
```
could you add this part also to the manifests for the Postgresql CRD? [Here](https://github.com/zalando/postgres-operator/blob/master/manifests/postgresql.crd.yaml) and [there](https://github.com/zalando/postgres-operator/blob/master/charts/postgres-operator/crds/postgresqls.yaml).

bbsbb replied:
Done.
Also noticed that the custom resource defs follow alphabetical order and I hadn't respected that, so adjusted it as well.

FxKu commented:
should be documented here: https://github.com/zalando/postgres-operator/blob/master/docs/reference/cluster_manifest.md

bbsbb commented:
CRDs adjusted, documentation for the spec option added.

FxKu commented on docs/reference/cluster_manifest.md:
```diff
@@ -65,6 +65,10 @@ These parameters are grouped directly under  the `spec` key in the manifest.
   It should be a [Spilo](https://github.com/zalando/spilo) image. Optional.
 
+* **schedulerName**
+  allows choice of scheduling profile for pods. If no value is provided
+  assumed is K8s default - `default-scheduler`. Optional.
```
```suggestion
  specifies the scheduling profile for pods. If no value is provided
  K8s' `default-scheduler` will be used. Optional.
```

FxKu commented:
can you rebase with the master. I've just merged a fix for CRDs and also updated go modules. Ideally, only changes relevant for this PR should be contained (e.g. no change to makefile etc.).

bbsbb commented:
> can you rebase with the master. I've just merged a fix for CRDs and also updated go modules. Ideally, only changes relevant for this PR should be contained (e.g. no change to makefile etc.).

On it.

bbsbb commented:
- Removed Makefile EOL change & deps bumps for clean PR :x: 
- Adjusted documentation per suggestion. :heavy_check_mark: 
- Rebased to latest master & squashed :heavy_check_mark: 

Cheers, let me know if anything else is necessary.

FxKu commented on docs/reference/cluster_manifest.md:
```diff
@@ -65,6 +65,10 @@ These parameters are grouped directly under  the `spec` key in the manifest.
   It should be a [Spilo](https://github.com/zalando/spilo) image. Optional.
 
+* **schedulerName**
+  specifies the scheduling profile for pods. If no value is provided
```
```suggestion
  specifies the scheduling profile for database pods. If no value is provided
```

FxKu commented:
Did one minor update as the scheduler name is only set for the database pods.

FxKu commented:
:+1:

Jan-M commented:
:+1:

FxKu commented:
:+1:
